packages/react-components/src/components/knowledge-graph/KnowledgeGraphPanel.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useCallback, useMemo, useState } from 'react';
+import { ElementDefinition } from 'cytoscape';
+import { Button, Container, Header, Input, SpaceBetween } from '@cloudscape-design/components';
+import { TwinMakerKGQueryDataModule } from '@iot-app-kit/source-iottwinmaker';
+import { Graph } from './graph';
+import StateManager, { useKnowledgeGraphState } from './StateManager';
+import { createKnowledgeGraphQueryClient } from './KnowledgeGraphQueries';
+import { ResponseParser } from './responseParser';
+import { getElementsDefinition } from './utils';
+
+interface KnowledgeGraphInterface {
+  kgDataSource: TwinMakerKGQueryDataModule;
+}
+const MAX_NUMBER_HOPS = 10;
+
+const KnowledgeGraphContainer: React.FC<KnowledgeGraphInterface> = ({ kgDataSource }) => {
+  const { selectedGraphNodeEntityId, setQueryResult, queryResult, clearGraphResults } = useKnowledgeGraphState();
+  const [searchTerm, setSearchTerm] = useState('');
+  //const [numOfHops, setNumOfHops] = useState(MAX_NUMBER_HOPS); // number of hops can be changed in settings which will come in the coming PRs.
+  const [elements, setElements] = useState<ElementDefinition[]>([]);
+
+  const knowledgeGraphQueryClient = useMemo(() => {
+    return createKnowledgeGraphQueryClient(kgDataSource, setQueryResult);
```

Comment:
I feel like the logic of all predefined and supported queries and initializing such a client should be a separate logic and not part of state management. it will influence state but it is in its own separate layer.

Reply:
graph.ts is the presentational components and doesn't have to know anything about data source or how data looks like. You just pass it nodes and edges and styles and it will draw them for you and thats exactly whats happening here.

Reply:
After looking at event handling in cytoscape I am merging the KnowledgeGraphPanel.tsx and graph.tsx in one file it seems there is an extra component/layer that is not needed. I will do that in the coming PR with event handling. And I will consider your suggestion here and evaluate after merging these 2 components into one.